src/test/rpc/NoRippleCheck_test.cpp

```diff
@@ -0,0 +1,328 @@
+            params[jss::role] = "user";
+            params[jss::ledger] = "current";
+            auto const result = env.rpc ("json", "noripple_check",
+                boost::lexical_cast<std::string>(params)) [jss::result];
+            BEAST_EXPECT (result[jss::error] == "badSeed");
+            BEAST_EXPECT (result[jss::error_message] ==
+                "Disallowed seed.");
+        }
+    }
+
+    void
+    testBasic (bool user, bool problems)
+    {
+        testcase << "Request noripple_check for " <<
+            (user ? "user" : "gateway") << " role, expect" <<
+            (problems ? "" : " no") << " problems";
+
+        using namespace test::jtx;
+        Env env {*this};
+
+        auto const gw = Account {"gw"};
+        auto const alice = Account {"alice"};
+
+        env.fund (XRP(10000), gw, alice);
+        if ((user && problems) || (!user && !problems))
+        {
+            env (fset (alice, asfDefaultRipple));
+            env (trust (alice, gw["USD"](100)));
+        }
+        else
+        {
+            env (fclear (alice, asfDefaultRipple));
+            env (trust (alice, gw["USD"](100), gw, tfSetNoRipple));
+        }
+        env.close ();
+
+        Json::Value params;
+        params[jss::account] = alice.human();
+        params[jss::role] = user ? "user" : "gateway";
+        params[jss::ledger] = "current";
+        auto result = env.rpc ("json", "noripple_check",
+            boost::lexical_cast<std::string>(params)) [jss::result];
+
+        auto const pa = result["problems"];
+        if (! BEAST_EXPECT (pa.isArray ()))
+            return;
+
+        if (problems)
+        {
+            if (! BEAST_EXPECT (pa.size() == 2))
+                return;
+
+            if (user)
+            {
+                BEAST_EXPECT (
+                    boost::starts_with(pa[0u].asString(),
+                        "You appear to have set"));
+                BEAST_EXPECT (
+                    boost::starts_with(pa[1u].asString(),
+                        "You should probably set"));
+            }
+            else
+            {
+                BEAST_EXPECT (
+                    boost::starts_with(pa[0u].asString(),
+                        "You should immediately set"));
+                BEAST_EXPECT(
+                    boost::starts_with(pa[1u].asString(),
+                        "You should clear"));
+            }
+        }
+        else
+        {
+            BEAST_EXPECT (pa.size() == 0);
+        }
+
+        // now make a second request asking for the relevant transactions this
+        // time.
+        params[jss::transactions] = true;
+        result = env.rpc ("json", "noripple_check",
+            boost::lexical_cast<std::string>(params)) [jss::result];
+        if (! BEAST_EXPECT (result[jss::transactions].isArray ()))
+            return;
+
+        auto const txs = result[jss::transactions];
+        if (problems)
+        {
+            if (! BEAST_EXPECT (txs.size () == user ? 1 : 2))
```

Comment:
I think the [precedence](http://en.cppreference.com/w/cpp/language/operator_precedence) of the ternary conditional is lower than the equality operator; extra parenthesis needed? 

VS gave warnings
```
NoRippleCheck_test.cpp(198): warning C4805: '==': unsafe mix of type 'Json::Value::UInt' and type 'bool' in operation
```